src/latch_cli/nextflow/parse_schema.py

```diff
@@ -0,0 +1,422 @@
+import json
+from pathlib import Path
+from textwrap import indent
+from typing import Generic, Literal, Optional, TypedDict, TypeVar, Union
+
+import click
+from typing_extensions import NotRequired, TypeAlias
+
+# todo(ayush): fix typing for defaults
+# todo(ayush): lots of type errors here, don't think any are that serious but would like to fix eventually
+
+
+class SchemaParsingError(Exception):
```

Comment:
```suggestion
class SchemaParsingError(RuntimeError):
```
usually dont want to inherit just `Exception`